### PR TITLE
Fix Tar Shot

### PR DESCRIPTION
**Pull request by KrisXV**
Tar Shot changes the target's effectiveness to become weaker against Fire-type moves instead of a set modifier.

### CODE REVIEW

File: data/moves.js [hunk 1]
```diff
@@ -19279,11 +19279,10 @@ let BattleMovedex = {
...
 				this.add('-start', pokemon, 'Tar Shot');
 			},
-			onSourceModifyDamage(damage, source, target, move) {
-				if (target.volatiles['tarshot'] && this.dex.getActiveMove(move).type === 'Fire') {
-					// TODO: Figure out damage modifier
-					return this.chainModify(2);
-				}
+			onEffectiveness(typeMod, target, type, move) {
+				if (move.type !== 'Fire') return;
+				if (!target || !target.volatiles['tarshot']) return;
```
Comment (Zarel) [inline data/moves.js]: I'm curious what situations this would be true in.

Comment (KrisXV) [reply data/moves.js]: i used the code for thousand arrows as a base and it used it as a crash prevention in chat commands. ig we dont really need the `!target` part

Comment (Zarel) [reply data/moves.js]: Do we need the other part?

Comment (urkerab) [reply data/moves.js]: We don't need either. This is running as an effect, so must be in a battle, while Thousand Arrows and Flying Press have the event directly on the move itself, which is why Thousand Arrows needs the chat command protection.

File: data/moves.js [hunk 2]
```diff
@@ -19279,11 +19279,10 @@ let BattleMovedex = {
...
+				return this.dex.getEffectiveness('Fire', target) + 1;
```
Comment (urkerab) [inline data/moves.js]: Could this be just `typeMod + 1`?

**Pull request merged**